arangod/Replication2/MetricsHelper.h

```diff
@@ -59,7 +58,11 @@ struct GaugeScopedCounter {
   void fire() noexcept;
 
  private:
-  std::shared_ptr<metrics::Gauge<std::uint64_t>> _metric;
+  struct noop {
+    template<typename T>
+    void operator()(T*) {}
+  };
```

Comment:
Why? You already delete copy ctor/assign. Why not raw pointer??
Also unique_ptr by default doesn't have trivial abi

Reply:
You are incorrect. Raw pointer don't provide the same move semantic. `fire` should trigger exactly once, and if the type is moved away, it should not trigger, but only if the moved-to object is destroyed. Thus a non-trivial ABI is required.

Reply:
1. Your object is already have move only semantics
2. In fire you manually call .reset(), and manually check nullptr.

This code will be completely same (=nullptr instead of .reset) without unique_ptr

Reply:
> non-trivial ABI is required

Any code where unique_ptr and pointer is equivalent not trivial ABI required

Reply:
No it will not be the same, because the move constructor/assignment operator differs.

Reply:
Ok, I see now, u use default move, sorry.